src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -0,0 +1,1255 @@
+        if (FAILED(settingsLoadedResult))
+        {
+            const winrt::hstring titleKey = USES_RESOURCE(L"InitialJsonParseErrorTitle");
+            const winrt::hstring textKey = USES_RESOURCE(L"InitialJsonParseErrorText");
+            _ShowLoadErrorsDialog(titleKey, textKey, settingsLoadedResult, _initialLoadResult.ExceptionText());
+        }
+        else if (settingsLoadedResult == S_FALSE)
+        {
+            _ShowLoadWarningsDialog(_initialLoadResult.Warnings());
+        }
+    }
+
+    // Method Description:
+    // - Helper for determining if the "Touch Keyboard and Handwriting Panel
+    //   Service" is enabled. If it isn't, we want to be able to display a
+    //   warning to the user, because they won't be able to type in the
+    //   Terminal.
+    // Return Value:
+    // - true if the service is enabled, or if we fail to query the service. We
+    //   return true in that case, to be less noisy (though, that is unexpected)
+    bool TerminalWindow::_IsKeyboardServiceEnabled()
+    {
+        if (IsUwp())
+        {
+            return true;
+        }
+
+        // If at any point we fail to open the service manager, the service,
+        // etc, then just quick return true to disable the dialog. We'd rather
+        // not be noisy with this dialog if we failed for some reason.
+
+        // Open the service manager. This will return 0 if it failed.
+        wil::unique_schandle hManager{ OpenSCManagerW(nullptr, nullptr, 0) };
+
+        if (LOG_LAST_ERROR_IF(!hManager.is_valid()))
+        {
+            return true;
+        }
+
+        // Get a handle to the keyboard service
+        wil::unique_schandle hService{ OpenServiceW(hManager.get(), TabletInputServiceKey.data(), SERVICE_QUERY_STATUS) };
+
+        // Windows 11 doesn't have a TabletInputService.
+        // (It was renamed to TextInputManagementService, because people kept thinking that a
+        // service called "tablet-something" is system-irrelevant on PCs and can be disabled.)
+        if (!hService.is_valid())
+        {
+            return true;
+        }
+
+        // Get the current state of the service
+        SERVICE_STATUS status{ 0 };
+        if (!LOG_IF_WIN32_BOOL_FALSE(QueryServiceStatus(hService.get(), &status)))
+        {
+            return true;
+        }
+
+        const auto state = status.dwCurrentState;
+        return (state == SERVICE_RUNNING || state == SERVICE_START_PENDING);
+    }
+
+    // Method Description:
+    // - Get the size in pixels of the client area we'll need to launch this
+    //   terminal app. This method will use the default profile's settings to do
+    //   this calculation, as well as the _system_ dpi scaling. See also
+    //   TermControl::GetProposedDimensions.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - a point containing the requested dimensions in pixels.
+    winrt::Windows::Foundation::Size TerminalWindow::GetLaunchDimensions(uint32_t dpi)
+    {

```

Comment:
Basically, for all these, it's not TerminalWindow's job anymore. TerminalWindow gets the liberty of making the assumption that the settings were already loaded. TerminalWindow takes a settings object in the ctor - this would never be hit :)